src/decisionengine/framework/engine/de_client.py

```diff
@@ -102,8 +107,13 @@ def execute_command_from_args(argsparsed, de_socket):
         return de_socket.stop()
     if argsparsed.print_engine_loglevel:
         return de_socket.get_log_level()
+    if argsparsed.product_dependencies:
+        return de_socket.product_dependencies()
     if argsparsed.block_while:
-        return de_socket.block_while(argsparsed.block_while, argsparsed.timeout)
+        timeout = argsparsed.timeout
+        if timeout is not None:
+            timeout = int(timeout)
```

Comment:
Why make an int?
I think the timeout can be a float as well, the partial value will become float because of how the time is counted.